x-pack/platform/plugins/shared/stack_connectors/public/connector_types/thehive/template_component/index.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState } from 'react';
+import {
+  EuiPopover,
+  EuiButtonIcon,
+  EuiText,
+  EuiSelectable,
+  EuiSpacer,
+  EuiHighlight,
+  useEuiTheme,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiToolTip,
+  EuiSelectableOption,
+} from '@elastic/eui';
+import { TruncatedText } from './truncated_text';
+import * as i18n from './translations';
+
+interface Props {
+  buttonTitle: string;
+  paramsProperty: string;
+  onSelectEventHandler: (template: Template) => void;
+}
+
+export interface Template {
+  name: string;
+  description: string;
+}
+
+const templates: Template[] = [
+  {
+    name: i18n.CUSTOM_TEMPLATE_LABEL,
+    description: i18n.CUSTOM_TEMPLATE_DESCRIPTION,
+  },
+  {
+    name: i18n.COMPROMISED_USER_ACCOUNT_INVESTIGATION_LABEL,
+    description: i18n.COMPROMISED_USER_ACCOUNT_INVESTIGATION_DESCRIPTION,
+  },
+  {
+    name: i18n.MALICIOUS_FILE_ANALYSIS_LABEL,
+    description: i18n.MALICIOUS_FILE_ANALYSIS_DESCRIPTION,
+  },
+  {
+    name: i18n.SUSPICIOUS_NETWORK_ACTIVITY_LABEL,
+    description: i18n.SUSPICIOUS_NETWORK_ACTIVITY_DESCRIPTION,
+  },
+];
+
+export const TemplateOptions: React.FunctionComponent<Props> = ({
+  buttonTitle,
+  paramsProperty,
+  onSelectEventHandler,
+}) => {
+  const [isTemplatesPopoverOpen, setIsTemplatesPopoverOpen] = useState<boolean>(false);
+
+  const { euiTheme } = useEuiTheme();
+
+  const templatesObject: Record<string, Template> = {};
+  templates?.forEach((template) => {
+    templatesObject[template.name] = template;
+  });
+
+  const optionsToShow = templates?.map((template) => ({
+    label: template.name,
+    data: {
+      description: template.description,
+    },
+    'data-test-subj': `${template.name}-selectableOption`,
+  }));
+
+  const Button = useMemo(
+    () => (
+      <EuiButtonIcon
+        id={`${paramsProperty}AddVariableButton`}
+        data-test-subj={`${paramsProperty}TemplateSelectButton`}
+        title={buttonTitle}
+        onClick={() => setIsTemplatesPopoverOpen(!isTemplatesPopoverOpen)}
+        iconType="documents"
+        aria-label={buttonTitle}
+      />
+    ),
+    [buttonTitle, isTemplatesPopoverOpen, paramsProperty]
+  );
+
+  const ToolTipContent = ({ description, label }: { description: string; label: string }) => {
+    return (
+      <>
+        <EuiText
+          size="s"
+          style={{
+            fontWeight: euiTheme.font.weight.bold,
+          }}
+        >
+          {label}
+        </EuiText>
+        <EuiSpacer size="s" />
+        <hr />
+        <EuiSpacer size="s" />
+        <EuiText size="xs">{description}</EuiText>
+      </>
+    );
+  };
```

Comment:
Could we extract these two guys as separate components? It's better not to define components inside components.